pkg/services/secrets/kvstore/migrate_from_plugin.go

```diff
@@ -0,0 +1,94 @@
+package kvstore
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin/secretsmanagerplugin"
+	"github.com/grafana/grafana/pkg/services/secrets"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+// MigrateFromPluginService This migrator will handle migration of the configured plugin secrets back to Grafana unified secrets
+type MigrateFromPluginService struct {
+	secretsStore   SecretsKVStore
+	cfg            *setting.Cfg
+	logger         log.Logger
+	sqlStore       sqlstore.Store
+	secretsService secrets.Service
+	manager        plugins.SecretsPluginManager
+}
+
+func ProvideMigrateFromPluginService(
+	secretsStore SecretsKVStore,
+	sqlStore sqlstore.Store,
+	secretsService secrets.Service,
+	manager plugins.SecretsPluginManager,
+) *MigrateFromPluginService {
+	return &MigrateFromPluginService{
+		secretsStore:   secretsStore,
+		logger:         log.New("sec-plugin-mig"),
+		sqlStore:       sqlStore,
+		secretsService: secretsService,
+		manager:        manager,
+	}
+}
+
+func (s *MigrateFromPluginService) Migrate(ctx context.Context) error {
+	s.logger.Debug("starting migration of plugin secrets to unified secrets")
+	// access the plugin directly
+	plugin, err := startAndReturnPlugin(s.manager, context.Background())
```

Comment:
In this context, I think we only use the plugin to perform the migration, and it should not be used again. Do we have a way of shutting it down? It would be a process that is kept alive until a grafana restart.

Reply:
added a call to stop at the end of the operation